### PR TITLE
add ini-nested parser

### PR DESCRIPTION
**Pull request by muescha**
TODO:
- [ ] add docs
- [ ] add tests for parser
- [ ] add test for utils
- [ ] add more fixtures

from:
- Feature Request: Support [nested][groups] in ini parser #336

```ini
[Containments][1][Applets][26]
immutability=1
plugin=org.kde.plasma_applet_dict

[Containments][1][Applets][26][Configuration][ConfigDialog]
DialogHeight=539
DialogWidth=719

[Containments][1][Applets][26][Configuration][General]
dictionary=all,fd-eng-rus

[Containments][1][General]
ToolBoxButtonState=topcenter
ToolBoxButtonX=360
positions={"1707x960":[]}
sortMode=-1
```

should generate nested groups:

```json
{
  "Containments": [
    {
      "Applets": [
        {
          "immutability": "1",
          "plugin": "org.kde.plasma_applet_dict",
          "Configuration": {
            "ConfigDialog": {
              "DialogHeight": "539",
              "DialogWidth": "719"
            },
            "General": {
              "dictionary": "all,fd-eng-rus"
            }
          }
        }
      ],
      "General": {
        "ToolBoxButtonState": "topcenter",
        "ToolBoxButtonX": "360",
        "positions": "{\"1707x960\":[]}",
        "sortMode": "-1"
      }
    }
  ]
}

```

### CODE REVIEW

File: jc/utils.py
Comment (kellyjonbrazil) [inline jc/utils.py:1]: I'm not sure that these will be often-enough used functions to put them in `utils.py`.

File: tests/fixtures/generic/ini_nested__plasmarc.json [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      "activityId": "",
+      "formfactor": "2",
+      "immutability": "1",
+      "lastScreen": "0",
+      "location": "4",
+      "plugin": "org.kde.plasma.private.systemtray",
+      "popupHeight": "431",
+      "popupWidth": "431",
+      "wallpaperplugin": "org.kde.image",
+      "Applets": [
```
Comment (kellyjonbrazil) [inline tests/fixtures/generic/ini_nested__plasmarc.json:151]: It seems like information is being lost in the conversion. Should there be some sort of `_item` key to provide the original index number?

**Review by kellyjonbrazil — COMMENTED** [review]
Thanks for the parser PR!

**Comment by muescha** [comment]
I think this PR was auto closed by deleting the `dev` branch. 

I guess I can resubmit it to the new dev branch?

**Comment by kellyjonbrazil** [comment]
I reopened on the new `dev` but there are a lot of conflicts so it may be better to just cherry pick your original commits to the new branch.

**Comment by kellyjonbrazil** [comment]
Honestly I'd just stash/copy your `ini_dup.py` and test files. Then do another fork and copy those files over. I didn't realize a rebase would change all of the commit hashes so it ruined the branch. My bad! 😦

**Comment by muescha** [comment]
I work on it - i will open a new PR when ready :)

**Pull request closed without merging**